# Pandas 
Pandas is a popular python library for working with data through a `DataFrame`

In [ ]:
import pandas as pd

Read the csv file using the pandas method **read_csv** which returns a **DataFrame** object

In [6]:
raw_data_csv_file_name = 'data/fer2013.csv'
raw_data = pd.read_csv(raw_data_csv_file_name)

## **info()** for a quick description

In [7]:
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35887 entries, 0 to 35886
Data columns (total 3 columns):
emotion    35887 non-null int64
pixels     35887 non-null object
Usage      35887 non-null object
dtypes: int64(1), object(2)
memory usage: 841.2+ KB


35887 entries

3 Columns
* pixels: **object**
* emotion: **int64**
* Usage: **object**

## head() for first 5 rows

In [11]:
raw_data.head()

emotion                                             pixels     Usage
0        0  70 80 82 72 58 58 60 63 54 58 60 48 89 115 121...  Training
1        0  151 150 147 155 148 133 111 140 170 174 182 15...  Training
2        2  231 212 156 164 174 138 161 173 182 200 106 38...  Training
3        4  24 32 36 30 32 23 19 20 30 41 21 22 32 34 21 1...  Training
4        6  4 0 0 0 0 0 0 0 0 0 0 0 3 15 23 28 48 50 58 84...  Training

* emotion: 0-6
* pixels: "ints separated by spaces" 
* Usage: "Training" // maybe others

## value_counts() for value counts

In [9]:
raw_data["Usage"].value_counts()

Training       28709
PrivateTest     3589
PublicTest      3589
Name: Usage, dtype: int64